src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -270,6 +271,20 @@ void OTARequestor::ConnectToProvider(OnConnectedAction onConnectedAction)
                    ChipLogError(SoftwareUpdate, "Cannot establish connection to provider: %" CHIP_ERROR_FORMAT, err.Format()));
 }
 
+// Application directs the Requestor to cancel image update in progress. All the Requestor state is
+// cleared, UpdateState is reset to Idle
+void OTARequestor::CancelImageUpdate()
+{
+    mBdxDownloader->EndDownload(CHIP_ERROR_CONNECTION_ABORTED);
+
+    // All Provider info should be invalidated
+    mProviderNodeId      = kUndefinedNodeId;
+    mProviderFabricIndex = kUndefinedFabricIndex;
+    mProviderEndpointId  = kRootEndpointId;
+
+    RecordNewUpdateState(OTAUpdateStateEnum::kIdle, OTAChangeReasonEnum::kUnknown);
```

Comment:
Should we actually call `RecordErrorUpdateState` so that `HandleError` can be called on the driver?

Reply:
I don't know. Are we in an error situation here? The application decides to cancel the image update process, the core OTA logic successfully stops the download and resets all the state. I don't necessarily see that any of this constitutes an error and requires error handling from the platform.

Reply:
This more or less relates to the topic we had yesterday on periodic query. If we were to cancel, the entire process ends with this function. If we were to call `HandleError` (that function is currently empty), perhaps we could handle a restart of query after X time if we were to pass in the right `UpdateFailureState` here. Thoughts?

Reply:
Yes, we may decide to pass some sort of notification to the driver but it would have to be something tailored to this particular situation -- "Cancel Retries" or something like that, not a general error event.

Reply:
Could you add the API to the driver? Don't necessarily have to implement it.

Reply:
Done